root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -498,6 +498,41 @@ const {LINK_TYPES, cleanURL, guessType, validationRules} = require('../../edit/U
                     expected_clean_url: 'https://www.beatport.com/release/pryda-10-vol-i/1563118',
                only_valid_entity_types: ['release']
         },
+        {                               // Used to fool the detection of missing slug (MBS-9846)
+                             input_url: 'http://classic.beatport.com/release/4/2361374',
+                     input_entity_type: 'release',
+            expected_relationship_type: 'downloadpurchase',
+                    expected_clean_url: 'https://www.beatport.com/release/4/2361374',
+               only_valid_entity_types: ['release']
+        },
+        {                               // Legacy URL format (real example)
+                             input_url: 'https://www.beatport.com/en-US/html/content/release/detail/161035/Back%20To%20The%20Future',
+                     input_entity_type: 'release',
+            expected_relationship_type: 'downloadpurchase',
+                    expected_clean_url: 'https://www.beatport.com/release/back-to-the-future/161035',
+               only_valid_entity_types: ['release']
+        },
+        {                               // Legacy URL format (made up to test slug conversion)
+                             input_url: 'https://www.beatport.com/en-US/html/content/release/detail/06130/%40@%26%25%24$%23%22%21!-&tracks#',
```

Comment:
Still a 400 error? (Not sure if intentional.)

Reply:
Not intentional but it doesn’t matter as it is a made up example that spares about 10 real examples (about 1 per character, that can be found on Beatport). It is only supposed to cover the conversion of all reserved characters at once (as well as removing leading/repeated/trailing hyphens), not necessarily to resolve on Beatport.